dind/Dockerfile

```diff
@@ -1,26 +1,35 @@
 ARG DOCKER_VERSION
 
-FROM docker:${DOCKER_VERSION}-dind-rootless
+FROM golang:alpine AS build
+WORKDIR /build
+COPY docker-iptables-glue/go.* ./
+RUN go mod download
+COPY docker-iptables-glue/ ./
+RUN CGO_ENABLED=0 GOOS=linux go build
 
+FROM docker:${DOCKER_VERSION}-dind-rootless
 USER root
 
-RUN export CRUN_LATEST=$(wget -qO- https://api.github.com/repos/containers/crun/releases/latest | sed -En 's/.*"tag_name"\s*:\s*"((\\"|[^"])*)".*/\1/p') && \
-    wget -qO /usr/local/bin/crun "https://github.com/containers/crun/releases/download/${CRUN_LATEST}/crun-${CRUN_LATEST}-linux-amd64" && \
-    chmod +x /usr/local/bin/crun && \
-    export SLIRP4NETNS_LATEST=$(wget -qO- https://api.github.com/repos/rootless-containers/slirp4netns/releases/latest | sed -En 's/.*"tag_name"\s*:\s*"((\\"|[^"])*)".*/\1/p') && \
-    wget -qO /usr/local/bin/slirp4netns "https://github.com/rootless-containers/slirp4netns/releases/download/${SLIRP4NETNS_LATEST}/slirp4netns-x86_64" && \
-    chmod +x /usr/local/bin/slirp4netns && \
-    mkdir /opt/containerd && \
-    chown rootless:rootless /opt/containerd && \
-    mkdir /run/netns && \
-    chown rootless:rootless /run/netns
+ENV CRUN_BASE_DOWNLOAD_PATH=https://github.com/containers/crun/releases/download \
+    SLIRP4NETNS_BASE_DOWNLOAD_PATH=https://github.com/rootless-containers/slirp4netns/releases/download
```

Comment:
By doing that you are persisting the environment variables to the runtime. I think it's better to just leave it in place.

Reply:
aeab21d